content-review/impl/compilatio/src/main/java/org/sakaiproject/contentreview/compilatio/CompilatioReviewServiceImpl.java

```diff
@@ -0,0 +1,1084 @@
+
+			//TEMPORARY_UNAVAILABLE
+			if (document == null) {
+				return false;
+			}
+			
+			Element root = document.getDocumentElement();
+			
+			String externalId = null;
+			if (root.getElementsByTagName("idDocument").item(0) != null) {
+				externalId = getNodeValue("idDocument", root);
+			}
+			
+			if (externalId != null) {
+				if (externalId.length() > 0) {
+					log.debug("Submission successful");
+					currentItem.setExternalId(externalId);
+					currentItem.setStatus(ContentReviewConstants.CONTENT_REVIEW_NOT_SUBMITTED_CODE);
+					currentItem.setRetryCount(Long.valueOf(0));
+					currentItem.setLastError(null);
+					currentItem.setErrorCode(null);
+					currentItem.setDateSubmitted(new Date());
+					crqs.update(currentItem);
+				} else {
+					log.warn("invalid external id");
+					processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_CODE, "Submission error: no external id received", null);
+					return false;
+				}
+			} else {
+				String rMessage = null, rCode = null;
+
+				if (root.getElementsByTagName("faultstring").item(0) != null && root.getElementsByTagName("faultcode").item(0) != null) {
+					rMessage = getNodeValue("faultstring", root);
+					rCode = getNodeValue("faultcode", root);
+				}
+				
+				log.debug("Add Document To compilatio not successful: " + rMessage + "(" + rCode + ")");
+				int errorCodeInt = -1;
+				CompilatioError errorCode = CompilatioError.valueOf(rCode);
+				if (errorCode != null) {
+					errorCodeInt = errorCode.getErrorCode();
+				}
+				processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_CODE, "Add Document To compilatio Error: " + rMessage + "(" + rCode + ")", errorCodeInt);
+				return false;
+			}
+		} catch (Exception e) {
+			processError(currentItem, ContentReviewConstants.CONTENT_REVIEW_SUBMISSION_ERROR_RETRY_CODE, "Error Submitting Assignment for Submission: " + e.getMessage() + ". Assume unsuccessful", null);
+			return false;
+		}
+
+		return true;
+	}
+	
+	public String escapeFileName(String fileName, String contentId) {
+		log.debug("original filename is: " + fileName);
+		if (fileName == null) {
+			// use the id
+			fileName = contentId;
+		} else if (fileName.length() > 199) {
+			fileName = fileName.substring(0, 199);
+		}
+		log.debug("fileName is :" + fileName);
+		try {
+			fileName = URLDecoder.decode(fileName, "UTF-8");
+			// in rare cases it seems filenames can be double encoded
+			while (fileName.indexOf("%20") > 0 || fileName.contains("%2520")) {
+				try {
+					fileName = URLDecoder.decode(fileName, "UTF-8");
+				} catch (IllegalArgumentException eae) {
+					log.warn("Unable to decode fileName: " + fileName);
+					eae.printStackTrace();
+					// as the result is likely to cause a MD5 exception use the
+					// ID
+					return contentId;
+				}
+
+			}
+		} catch (IllegalArgumentException eae) {
+			log.warn("Unable to decode fileName: " + fileName);
+			eae.printStackTrace();
```

Comment:
Don't print the stacktrace, just pass the exception object to your logger statement above